### PR TITLE
Add quick-lint-js version 2.4.1

### PR DESCRIPTION
**Pull request by strager**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/55794)

### CODE REVIEW

**Comment by strager** [comment]
> Hello @strager, The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

I don't see what the issue is. Here is the raw log from the [failing job](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=296620&view=logs&j=871b4eed-22de-5cb6-c1c8-82f190de633f&t=58bdf892-4e45-5ba7-4782-31912f95e167):
```

3/26/2022 12:53:44 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: Started
============================================================================== 
Task         : Installation Validation
Description  : Invoke an Azure Function
Version      : 1.198.0
Author       : Microsoft Corporation
Help URL     : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/azure-function
============================================================================== 
2022-03-26T00:53:46.1732738Z POST ***?code=***&
Request body: {
  "operationId": "WinGetSvc-Validation-213-55794-20220326-2",
  "BuildId": "296620",
  "PlanUrl": "https://dev.azure.com/ms/",
  "HubName": "build",
  "ProjectId": "ed6a5dfa-6e7f-413b-842c-8305dd9e89e6",
  "PlanId": "7609bccd-b454-468f-ab55-87bffa98f576",
  "JobId": "871b4eed-22de-5cb6-c1c8-82f190de633f",
  "TimelineId": "7609bccd-b454-468f-ab55-87bffa98f576",
  "TaskInstanceId": "58bdf892-4e45-5ba7-4782-31912f95e167",
  "AuthToken": ***
}
				Response Code: OK
				Response: 

3/26/2022 12:55:22 AM

****************************************************************************************
Installation Verification
****************************************************************************************

-----------------------------------------------------
Failed
-----------------------------------------------------
Status: Completed
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
```
2022-03-25 17:55:03.692 [CORE] Valid settings file not found. Using default values.
2022-03-25 17:55:03.692 [CORE] WinGet, version [1.1.12653], activity [{32DABFB0-D94A-4417-B495-4AF11CE667C0}]
2022-03-25 17:55:03.693 [CORE] OS: Windows.Desktop v10.0.19042.1052
2022-03-25 17:55:03.693 [CORE] Command line Args: C:\Users\validator\AppData\Local\Microsoft\WindowsApps\winget.exe install --manifest D:\TRANSFER\85daed34-9fac-ec11-b6f8-0022484bf5bb\Manifest.yaml --accept-package-agreements
2022-03-25 17:55:03.693 [CORE] Package: Microsoft.DesktopAppInstaller v1.16.12653.0
2022-03-25 17:55:03.695 [CORE] IsCOMCall:0; Caller: winget-cli
2022-03-25 17:55:03.719 [CLI ] WinGet invoked with arguments: 'install' '--manifest' 'D:\TRANSFER\85daed34-9fac-ec11-b6f8-0022484bf5bb\Manifest.yaml' '--accept-package-agreements'
2022-03-25 17:55:03.719 [CLI ] Found subcommand: install
2022-03-25 17:55:03.719 [CLI ] Leaf command to execute: root:install
2022-03-25 17:55:03.733 [CLI ] Executing command: install
2022-03-25 17:55:03.739 [CLI ] Manifest fields: Name [quick-lint-js], Version [2.3.1]
2022-03-25 17:55:03.742 [CLI ] Starting installer selection.
2022-03-25 17:55:03.742 [CLI ] Completed installer selection.
2022-03-25 17:55:03.767 [CLI ] Generated temp download path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1
2022-03-25 17:55:03.769 [CORE] Downloading to path: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1
2022-03-25 17:55:03.769 [CORE] DeliveryOptimization downloading from url: https://c.quick-lint-js.com/releases/2.3.1/windows/quick-lint-js.msix
2022-03-25 17:55:04.740 [CORE] Download completed.
2022-03-25 17:55:04.759 [CORE] Started applying motw to C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1 with zone: 3
2022-03-25 17:55:04.767 [CORE] Finished applying motw
2022-03-25 17:55:04.773 [CLI ] Installer hash verified
2022-03-25 17:55:04.774 [CORE] Started applying motw using IAttachmentExecute to C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1
2022-03-25 17:55:04.786 [CORE] Started removing motw to C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1
2022-03-25 17:55:04.787 [CORE] Finished removing motw
2022-03-25 17:55:05.089 [CORE] Finished applying motw using IAttachmentExecute. Result: 0 IAttachmentExecute::Save() result: 0
2022-03-25 17:55:05.232 [CORE] Starting RequestAddPackageAsync operation #0: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\quick-lint.quick-lint-js.2.3.1
2022-03-25 17:55:05.237 [CORE] Begin waiting for operation #0
2022-03-25 17:55:05.237 [CORE] Begin blocking for operation #0
2022-03-25 17:55:05.317 [CORE] Deployment operation #0: error 0x800B0109: The root certificate of the signature in the app package or bundle must be trusted.
2022-03-25 17:55:05.317 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(159)\AppInstallerCLI.exe!00007FF62962A3A4: (caller: 00007FF6295D746C) Exception(1) tid(6bc) 800B0109 A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider.


2022-03-25 17:55:05.318 [CORE] Starting RemovePackage operation #1: fa6b940b-8f96-4b46-8e21-703a63133e06_2.3.1.0_neutral__qgt2h5p98tnve
2022-03-25 17:55:05.338 [CORE] Begin waiting for operation #1
2022-03-25 17:55:05.338 [CORE] Begin blocking for operation #1
2022-03-25 17:55:05.454 [CORE] Deployment operation #1: Windows cannot remove fa6b940b-8f96-4b46-8e21-703a63133e06_2.3.1.0_neutral__qgt2h5p98tnve because the current user does not have that package installed. Use Get-AppxPackage to see the list of packages installed.
2022-03-25 17:55:05.454 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(54)\AppInstallerCLI.exe!00007FF629629124: (caller: 00007FF62962A6E5) Exception(2) tid(6bc) 80073CF1 Package was not found.

    Msg:[Operation failed: Windows cannot remove fa6b940b-8f96-4b46-8e21-703a63133e06_2.3.1.0_neutral__qgt2h5p98tnve because the current user does not have that package installed. Use Get-AppxPackage to see the list of packages installed.] 

2022-03-25 17:55:05.455 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(124)\AppInstallerCLI.exe!00007FF62977022E: (caller: 00007FF62962A783) LogHr(1) tid(6bc) 80073CF1 Package was not found.

    Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Deployment.cpp(54)\AppInstallerCLI.exe!00007FF629629124: (caller: 00007FF62962A6E5) Exception(2) tid(6bc) 80073CF1 Package was not found.

    Msg:[Operation failed: Windows cannot remove fa6b940b-8f96-4b46-8e21-703a63133e06_2.3.1.0_neutral__qgt2h5p98tnve because the current user does not have that package installed. Use Get-AppxPackage to see the list of packages installed.] 
] 

2022-03-25 17:55:05.488 [CLI ] MSIX installer failed: 2148204809
2022-03-25 17:55:05.492 [CLI ] Terminating context: 0x800b0109 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\InstallFlow.cpp:23b
```

Logs can be found from https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=296620&view=artifacts&pathAsName=false&type=publishedArtifacts

I'm on mobile so I cannot check but there's two things that's compulsory in MSIX:
1) Self-signed (Authenticode) certificates cannot be used.
2) (Authenticode) certificate must be from a reputable company (i.e. Comodo is one example) where the company's root certificate is in the OS Trust Store.

^ Azure Pipelines is running Windows 10 version 20H2 with the latest CU that was released in June 2021.

EDIT: You should use winget-create to create your manifests which can be used for your CI/CD pipelines.

Hardcoding it in your code prevents you from taking advantage of newer WinGet features that are only available with latest schema versions and hardcoding it will may also break your pull requests in the future in case older schema versions have its support dropped.

**Comment by strager** [comment]
> there's two things that's compulsory in MSIX:
>
> * Self-signed (Authenticode) certificates cannot be used.
> * (Authenticode) certificate must be from a reputable company (i.e. Comodo is one example) where the company's root certificate is in the OS Trust Store.

Does winget require code signatures? It looks like some installers in winget are not Authenticode-signed, such as JanDeDobbeleer.OhMyPosh (but I might be wrong):
https://github.com/microsoft/winget-pkgs/blob/8252a7427bac01bfb90b11452b765add3a6c6cf8/manifests/j/JanDeDobbeleer/OhMyPosh/7.49.0/JanDeDobbeleer.OhMyPosh.installer.yaml#L17

**Comment by ItzLevvie** [comment]
> Does winget require code signatures?

WinGet follows whatever App Installer wants.

The MSIX specifications require you to purchase a Code Signing Certificate and get it cross-signed with a CA such as Comodo or 

> It looks like some installers in winget are not Authenticode-signed, such as JanDeDobbeleer.OhMyPosh

That is correct but signing is only applicable and required for APPX and MSIX installers.

It doesn't affect EXE / MSI / ZIP or other installer types.

If you don't want to purchase a CSC you can wait until WinGet supports `.zip` installer type or make an EXE or MSI installer.

**Comment by Trenly** [comment]
I would also like to add that the `SignatureSha256` should be included as well as the Yaml schema headers. I ran it through YamlCreate and got [this](https://github.com/trenly/winget-pkgs/commit/49b4d8221ff1302c22c6b180dc96767628070007).

Testing in the sandbox while using the manifests I got from YamlCreate gave this output, which confirms that even though the package is signed, it is not signed using a certificate from a provider that Microsoft trusts.
```
--> Installing WinGet
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Admin setting enabled.

--> Installing the Manifest 2.3.1

Found quick-lint-js [quick-lint.quick-lint-js] Version 2.3.1
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Installer failed with exit code: 0x800b0109 : A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider.
```

**Comment by strager** [comment]
> This is correct but authenticode signing is only required for APPX / MSIX installers. This doesn't affect EXE / MSI or other installer types because it's not compulsory.

Ah, okay. That makes sense.

I'll look into getting the MSIX certificate signed by a Microsoft-trusted authority.

**Comment by strager** [comment]
The MSIX installer is now signed with a Microsoft-trusted chain.

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# To the extent possible under law, Matthew "strager" Glazar
+# has waived all copyright and related or neighboring rights
+# to _quick-lint-js winget manifests_. This work is
+# published from United States.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+PackageIdentifier: quick-lint.quick-lint-js
+PackageVersion: 2.4.1
```
Comment (ItzLevvie) [inline manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml:8]:
```suggestion
PackageVersion: 2.4.1.0
```
This must match Control Panel.

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# To the extent possible under law, Matthew "strager" Glazar
+# has waived all copyright and related or neighboring rights
+# to _quick-lint-js winget manifests_. This work is
+# published from United States.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+PackageIdentifier: quick-lint.quick-lint-js
+PackageVersion: 2.4.1
```
Comment (ItzLevvie) [inline manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:8]:
```suggestion
PackageVersion: 2.4.1.0
```
This must match Control Panel.

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# To the extent possible under law, Matthew "strager" Glazar
+# has waived all copyright and related or neighboring rights
+# to _quick-lint-js winget manifests_. This work is
+# published from United States.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+PackageIdentifier: quick-lint.quick-lint-js
+PackageVersion: 2.4.1
```
Comment (ItzLevvie) [inline manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.yaml:8]:
```suggestion
PackageVersion: 2.4.1.0
```
This must match Control Panel.

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+PackageName: quick-lint-js
+PackageUrl: https://quick-lint-js.com/
+ShortDescription: 'Find bugs in JavaScript programs.'
+Moniker: quicklintjs
+
+Publisher: quick-lint
+PublisherSupportUrl: https://quick-lint-js.com/contact/
+Copyright: 'Copyright (C) 2020  Matthew "strager" Glazar'
+License: 'GPL-3.0-or-later,...'
```
Comment (ItzLevvie) [inline manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml:19]:
```suggestion
License: 'GPL-3.0-or-later'
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+Installers:
+- InstallerUrl: https://c.quick-lint-js.com/releases/2.4.1/windows/quick-lint-js.msix
+  InstallerSha256: 8601c5a6151991af730994bd048a59b8fe2e0fbb59fa9fb1eb67a4be6b41deb0
+  InstallerType: msix
+  Platform:
+  - Windows.Desktop
+  - Windows.Universal
+  MinimumOSVersion: 10.0.0.0
+  Architecture: x64
+  PackageFamilyName: fa6b940b-8f96-4b46-8e21-703a63133e06_y724facbts4te
```
Comment (ItzLevvie) [inline manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:19]:
```suggestion
  PackageFamilyName: fa6b940b-8f96-4b46-8e21-703a63133e06_116557qebcmn6
```
to match Control Panel.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml (hunk 5 above)
Comment (strager) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:19]: Where did you get this code from?

Comment (ItzLevvie) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:19]: I installed the MSIX and ran `$((Get-AppxPackage | Select-Object -Last 1).PackageFamilyName)` in PowerShell.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml (hunk 5 above)
Comment (ItzLevvie) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:19]: I installed the MSIX and ran `$((Get-AppxPackage | Select-Object -Last 1).PackageFamilyName)` in Windows PowerShell.

Comment (lechacon) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml:19]: As a note, the package family name is a combination of the package name and a hash of the publisher name, you can see both in your AppxManifest.xml. The publisher name matches the subject of the signing certificate, so you may have had to change it to sign it with a trusted certificate.

If you are creating the package for the first time, you may want to use a more descriptive package name.

Comment (strager) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.installer.yaml]: Thanks for the advice. I changed the name of the package.

File: manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml (hunk 4 above)
Comment (strager) [reply manifests/q/quick-lint/quick-lint-js/2.4.1/quick-lint.quick-lint-js.locale.en-US.yaml]: I included all the licenses for the bundled software. Let me know if this is a problem.

**Review by ItzLevvie — APPROVED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Pull request merged**